pkg/models/query.go

```diff
@@ -194,6 +194,12 @@ func ApplyDefaultsToQuery(ctx context.Context, query Query, settings InfinitySet
 			query.Source = "url"
 		}
 	}
+	// if the parser is already set, we should respect that
+	// if the root_selector is already set, overriding the parser type will break the queries with frontend parsing. So we should leave as it is
+	// if the query have columns defined, overriding the parser type will break the queries with frontend parsing. So we should leave as it is
+	if query.Parser == "" && strings.TrimSpace(query.RootSelector) == "" && len(query.Columns) == 0 && query.Type != QueryTypeUQL && query.Type != QueryTypeGROQ && query.Type != QueryTypeGSheets {
```

Comment:
```suggestion
	if query.Parser == "" && strings.TrimSpace(query.RootSelector) == "" && len(query.Columns) == 0 {
```

nit, but here I don't think we need to add `query.Type != QueryTypeUQL && query.Type != QueryTypeGROQ && query.Type != QueryTypeGSheets` check, I don't think not having it breaks anything, because even now, when you switch to UQL query, the parser param is still there. 
<img width="1479" alt="image" src="https://github.com/user-attachments/assets/88925cd9-32ed-4ae2-979f-7b78ecb19fca" />

I believe this makes code cleaner and we don't have to unnecessary keep here a list of queries that don't support parser.

Reply:
sounds good!